src/main/java/org/terasology/web/resources/worldMap/WorldMapResource.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources.worldMap;
+
+import org.terasology.registry.In;
+import org.terasology.web.resources.base.AbstractSimpleResource;
+import org.terasology.web.resources.base.ClientSecurityRequirements;
+import org.terasology.web.resources.base.ResourceAccessException;
+import org.terasology.web.resources.base.ResourceMethod;
+import org.terasology.web.resources.base.ResourcePath;
+import org.terasology.world.WorldProvider;
+
+import static org.terasology.web.resources.base.ResourceMethodFactory.createParameterlessMethod;
+
+public class WorldMapResource extends AbstractSimpleResource {
+
+    @In
+    private WorldProvider worldProvider;
+
+    @Override
+    protected ResourceMethod<Void, WorldMapMetadata> getGetMethod(ResourcePath path) throws ResourceAccessException {
+        return createParameterlessMethod(path, ClientSecurityRequirements.PUBLIC, Void.class,
+                (data, client) -> new WorldMapMetadata(worldProvider));
+    }
+
+    @Override
+    protected ResourceMethod<WorldMapInput, WorldMapMetadata> getPutMethod(ResourcePath path) throws ResourceAccessException {
```

Comment:
Semantically (see [HTTP/1.1 Method Definitions](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html)), only GET would be the proper choice - PUT is meant to modify the resource, and the map isn't actually being modified here.

However I can see why you made this choice (I suppose) - there is no way to parametrize a GET request in our resource system. So I'd say it's fine to keep it this way until we update the system to support it. After all in plain HTTP (opposed to this custom wrapper/abstraction layer for HTTP and WebSocket) a request made with the GET method *can* be parametrized, usually with the [query string](https://en.wikipedia.org/wiki/Query_string).